docs/content/tutorial/ingress-tutorial.md

```diff
@@ -0,0 +1,139 @@
+- If you hit the external IP address of the ingress-nginx, you get the default backend page:
+
+```
+curl http://35.200.150.175
+
+default backend - 404
+```
+
+## Deploying Function with ingress
+
+### Create a function
+
+We will create an environment, function and test that it works:
+
+```
+$ fission env create --name nodejs --image fission/node-env
+environment 'nodejs' created
+
+$ cat hello.js 
+module.exports = async function(context) {
+    return {
+        status: 200,
+        body: "Hello, Fission!\n"
+    };
+}
+
+$ fission fn create --name hello --env nodejs --code hello.js 
+function 'hello' created
+
+$ fission fn test --name hello
+Hello, Fission!
+
+```
+
+### Create a internal route
+
+Let's create a route which is not exposed via the ingress controller:
+
+```
+$ fission route create --url /ihello --function hello
+trigger '249838c9-9ae3-492a-bba1-b0464ae65671' created
+$ fission route list
+NAME                                 METHOD HOST URL     INGRESS FUNCTION_NAME
+249838c9-9ae3-492a-bba1-b0464ae65671 GET         /ihello false   hello
+```
+
+This route will be accessible at `http://$FISSION_ROUTER/ihello` but if tried to access on the ingress controller address `http://<INGRESS-CONTROLLER-EXTERNAL-IP>/ihello` you will get a default backend page. This is expected result as we did not create an ingress for this route.
+
+### Create a external route
+
+Now let's create a route which we will expose over ingress controller. We will create a route with `createingress` flag enabled:
+
+```
+$ fission route create --url /hello --function hello --createingress
+trigger '301b3cb0-5ac1-4211-a1ed-2b0ad9143e34' created
+$ 
+$ fission route list
+NAME                                 METHOD HOST URL     INGRESS FUNCTION_NAME
+249838c9-9ae3-492a-bba1-b0464ae65671 GET         /ihello false   hello
+301b3cb0-5ac1-4211-a1ed-2b0ad9143e34 GET         /hello  true    hello
+$ 
+
+```
+If you check the ingress controller pod logs, you will notice that the ingress controller has re-loaded the configuration for the newly created ingress resource:
+
+```
+I0604 12:47:08.983567       5 controller.go:168] backend reload required
+I0604 12:47:08.985535       5 event.go:218] Event(v1.ObjectReference{Kind:"Ingress", Namespace:"fission", Name:"301b3cb0-5ac1-4211-a1ed-2b0ad9143e34", UID:"64bffe8c-67f5-11e8-98e8-42010aa00018", APIVersion:"extensions", ResourceVersion:"18017617", FieldPath:""}): type: 'Normal' reason: 'CREATE' Ingress fission/301b3cb0-5ac1-4211-a1ed-2b0ad9143e34
+I0604 12:47:09.117629       5 controller.go:177] ingress backend successfully reloaded...
+```
+
+If you now hit the function at ingress controller's IP and the path (`http://<INGRESS-CONTROLLER-EXTERNAL-IP>/hello`), you will get function response. Depending on your setup and settings, you will have to try HTTP or HTTPS. Some ingress controllers by default enable SSL redirect and hence HTTPS URL has to be accessed.
```

Comment:
Some ingress controllers by default enable SSL redirect and hence HTTPS URL has to be accessed. -> Some ingress controllers enable SSL redirect by default and hence the HTTPS URL has to be accessed.